go-controller/pkg/ovn/controller/apbroute/network_client.go

```diff
@@ -464,9 +472,10 @@ func (nb *northBoundClient) deletePodGWRoute(routeInfo *RouteInfo, podIP, gw, gr
 	}
 
 	node := util.GetWorkerFromGatewayRouter(gr)
+
 	// The gw is deleted from the routes cache after this func is called, length 1
 	// means it is the last gw for the pod and the hybrid route policy should be deleted.
-	if entry := routeInfo.PodExternalRoutes[podIP]; len(entry) == 1 {
+	if entry := routeInfo.PodExternalRoutes[podIP]; len(entry) <= 1 {
```

Comment:
what am I missing here, if entry is 0 already is there anything we need to delete? but I see your commit message saying we don't know exactly what happened and we are trying to be more careful, so I guess this means we try an extra delete to be sure we didn't miss a previous error etc.. that is fine for me but want to point out that `deletePodGWRoute` is always called only when entry is >=1. It's called either from:
```
for _, podIPNet := range podIPNets {
			podIP := podIPNet.IP.String()
			routes, ok := routeInfo.PodExternalRoutes[podIP]
			if !ok {
				continue
			}
			if len(routes) == 0 {
				delete(routeInfo.PodExternalRoutes, podIP)
				continue
			}
			for gw, gr := range routes {
				if !policyGWIPs.Has(gw) {
					if err := oc.deletePodGWRoute(routeInfo, podIP, gw, gr); err != nil {
						// if we encounter error while deleting routes for one pod; we return and don't try subsequent pods
						return fmt.Errorf("delete pod GW route failed: %w", err)
					}
					delete(routes, gw)
				}
			}
		}
```
OR
```
for podIP, routes := range routeInfo.PodExternalRoutes {
			for gw, gr := range routes {
				if (deleteAll || matchGWs.Has(gw)) && !policyGWIPs.Has(gw) {
					if err := oc.deletePodGWRoute(routeInfo, podIP, gw, gr); err != nil {
						// if we encounter error while deleting routes for one pod; we return and don't try subsequent pods
						return fmt.Errorf("delete pod GW route failed: %w", err)
					}
					delete(routes, gw)
				}
			}
		}
```
so we always loop through the routes, and if that is empty we would not reach here I guess.. but again I see no harm in doing this change per say.

Reply:
yeah, the original issue I hit in CI i wasnt sure how it happened, so I just thought I could make this change since it doesnt hurt